python/pyspark/sql/types.py

```diff
@@ -212,15 +213,32 @@ def needConversion(self) -> bool:
 
     def toInternal(self, dt: datetime.datetime) -> int:
         if dt is not None:
-            seconds = (
-                calendar.timegm(dt.utctimetuple()) if dt.tzinfo else time.mktime(dt.timetuple())
-            )
+            if platform.system().lower() == "windows":
```

Comment:
I'm just super surprised this is Windows specific?

Reply:
Yes, and I want you to check it out for yourself.